src/components/Store.react.js

```diff
@@ -2,27 +2,36 @@ import R from 'ramda';
 import React from 'react';
 import PropTypes from 'prop-types';
 
-function dataCheck(data, old) {
+/**
+ * Deep equality check to know if the data has changed.
+ *
+ * @param {any} newData - New data to compare
+ * @param {any} oldData - The old data to compare
+ * @returns {boolean} The data has changed.
+ */
+function dataChanged(newData, oldData) {
     // Assuming data and old are of the same type.
-    const oldNull = R.isNil(old);
-    const newNull = R.isNil(data);
+    const oldNull = R.isNil(oldData);
+    const newNull = R.isNil(newData);
     if (oldNull || newNull) {
         return oldNull !== newNull;
     }
-    const type = R.type(data);
+    const type = R.type(newData);
     if (type === 'Array') {
-        if (data.length !== old.length) {
+        if (newData.length !== oldData.length) {
             return true;
         }
-        for (let i = 0; i < data.length; i++) {
-            if (dataCheck(data[i], old[i])) {
+        for (let i = 0; i < newData.length; i++) {
+            if (dataChanged(newData[i], oldData[i])) {
                 return true;
             }
         }
     } else if (R.contains(type, ['String', 'Number'])) {
```

Comment:
`'Boolean'`?

Reply:
Forgot about `Boolean` in all this type checks, thanks!